### PR TITLE
Combine code of xiaomi_miscale and xiaomi_miscale2

### PR DESCRIPTION
**Pull request by edenhaus**
# What does this implement/fix? 

The code of `xiaomi_miscale` and `xiaomi_miscale2` is nearly the same.

The differences are:
- Miscale2 supports Impedance
- use different service UUID
- parsing of the actual data.
This PR combines the two platforms into one, as I think less duplicate code is easier to maintain.

As both version use different service uuid, we can indentify the version automatically on the service uuid.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#1458

## Test Environment

- [x] ESP32
- [ ] ESP8266

## Example entry for `config.yaml`:

```yaml

# For easier testing
external_components:
  - source: github://edenhaus/esphome@combine-xiaomi-scales
    components: [ xiaomi_miscale, xiaomi_miscale2, esp32_ble_tracker ]
    refresh: 0s

sensor:
  - platform: xiaomi_miscale
    mac_address: XX:XX:XX:XX:XX:XX # insert your mac address
    weight:
      name: Weight
    impedance:
      name: Impedance
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).
  
If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Comment by edenhaus** [comment]
@dckiller51 As you add the two platforms, can you please test and review it?
I only have a MiScale2 and there everything works correctly.

File: esphome/components/xiaomi_miscale/xiaomi_miscale.cpp [hunk 1]
```diff
@@ -80,17 +97,61 @@ bool XiaomiMiscale::parse_message(const std::vector<uint8_t> &message, ParseResu
...
+}
+
 bool XiaomiMiscale::report_results(const optional<ParseResult> &result, const std::string &address) {
   if (!result.has_value()) {
     ESP_LOGVV(TAG, "report_results(): no results available.");
     return false;
   }
 
-  ESP_LOGD(TAG, "Got Xiaomi Miscale (%s):", address.c_str());
+  ESP_LOGD(TAG, "Got Xiaomi Miscale v%d (%s):", (*result).version, address.c_str());
```
Comment (oxan) [inline esphome/components/xiaomi_miscale/xiaomi_miscale.cpp]:
```suggestion
  ESP_LOGD(TAG, "Got Xiaomi Miscale v%d (%s):", result->version, address.c_str());
```
For consistency.

File: esphome/components/xiaomi_miscale/xiaomi_miscale.cpp [hunk 2]
```diff
@@ -80,17 +97,61 @@ bool XiaomiMiscale::parse_message(const std::vector<uint8_t> &message, ParseResu
...
+  if (!is_stabilized || load_removed) {
+    return false;
+  }
+
+  // weight, 2 bytes, 16-bit  unsigned integer, 1 kg
+  const int16_t weight = uint16_t(data[11]) | (uint16_t(data[12]) << 8);
+  if (data[0] == 0x02)
+    result.weight = weight * 0.01f / 2.0f;  // unit 'kg'
+  else if (data[0] == 0x03)
+    result.weight = weight * 0.01f * 0.453592;  // unit 'lbs'
```
Comment (oxan) [inline esphome/components/xiaomi_miscale/xiaomi_miscale.cpp]:
```suggestion
    result.weight = weight * 0.01f * 0.453592f;  // unit 'lbs'
```

**Review by oxan — APPROVED** [review]
Thanks, nice cleanup!

**Comment by dckiller51** [comment]
> @ dckiller51 Au fur et à mesure que vous ajoutez les deux plates-formes, pouvez-vous le tester et le revoir ?
> Je n'ai qu'un MiScale2 et là tout fonctionne correctement.

Good initiative. I tests in the evening with the two. I will give you my results.

**Comment by dckiller51** [comment]
@edenhaus 

This works very well.
I think I had the same problem but v1 is always sending 

```
[00:38:49][D][xiaomi_miscale:147]: Got Xiaomi Miscale v2 (5C:XX:XX:XX:XX:XX):
[00:38:49][D][xiaomi_miscale:150]:   Weight: 77.55kg
[00:38:49][D][sensor:131]: 'Xiaomi Mi Scale Weight': Sending state 77.55000 kg with 2 decimals of accuracy
[00:38:54][D][xiaomi_miscale:147]: Got Xiaomi Miscale v2 (5C:XX:XX:XX:XX:XX):
[00:38:54][D][xiaomi_miscale:150]:   Weight: 77.55kg
[00:38:54][D][xiaomi_miscale:153]:   Impedance: 480ohm
[00:38:54][D][sensor:131]: 'Xiaomi Mi Scale Weight': Sending state 77.55000 kg with 2 decimals of accuracy
[00:38:54][D][sensor:131]: 'Xiaomi Mi Scale Impedance': Sending state 480.00000 Ω with 0 decimals of accuracy
[00:39:11][D][xiaomi_miscale:147]: Got Xiaomi Miscale v1 (C8:XX:XX:XX:XX:XX):
[00:39:11][D][xiaomi_miscale:150]:   Weight: 77.40kg
[00:39:11][D][sensor:131]: 'Weight Miscale1': Sending state 77.40000 kg with 2 decimals of accuracy
[00:39:12][D][xiaomi_miscale:147]: Got Xiaomi Miscale v1 (C8:XX:XX:XX:XX:XX):
[00:39:12][D][xiaomi_miscale:150]:   Weight: 77.40kg
[00:39:12][D][sensor:131]: 'Weight Miscale1': Sending state 77.40000 kg with 2 decimals of accuracy
[00:39:17][D][esp32_ble_tracker:628]: Found device 59:XX:XX:XX:XX:XX RSSI=-98
[00:39:17][D][esp32_ble_tracker:649]:   Address Type: RANDOM
[00:39:26][D][xiaomi_miscale:147]: Got Xiaomi Miscale v1 (C8:XX:XX:XX:XX:XX):
[00:39:26][D][xiaomi_miscale:150]:   Weight: 77.40kg
[00:39:26][D][sensor:131]: 'Weight Miscale1': Sending state 77.40000 kg with 2 decimals of accuracy
[00:39:51][D][xiaomi_miscale:147]: Got Xiaomi Miscale v1 (C8:XX:XX:XX:XX:XX):
[00:39:51][D][xiaomi_miscale:150]:   Weight: 77.40kg
[00:39:51][D][sensor:131]: 'Weight Miscale1': Sending state 77.40000 kg with 2 decimals of accuracy
```

```
esp32_ble_tracker:

sensor:
  - platform: xiaomi_miscale
    mac_address: C8:XX:XX:XX:XX:XX # insert your mac address
    weight:
      name: Weight Miscale1
  - platform: xiaomi_miscale
    mac_address: 5C:XX:XX:XX:XX:XX
    weight:
      name: "Xiaomi Mi Scale Weight"
    impedance:
      name: "Xiaomi Mi Scale Impedance"
```

**Comment by edenhaus** [comment]
> I think I had the same problem but v1 is always sending

I didn't change your code et all. It's still the same, I have only combined it into one platform.

For your problem with v1 you can use the build in sensor filters.
I would use a `throttle` combined with a `delta` filter:

```yaml
sensor:
  - platform: xiaomi_miscale
    mac_address: C8:XX:XX:XX:XX:XX # insert your mac address
    weight:
      name: Weight Miscale1
    filters:
      - or:
        - throttle: 10s
        - delta: 5.0
```

Above example will only pass forward values that are either at least 10s old or are if the absolute difference is at least 5.0.
Maybe a `debounce` filter is better

**Pull request merged**